### PR TITLE
[WIP] Separate time management from GameWindow

### PR DESCRIPTION
**Pull request by ghost**
### Purpose of this PR

* I wish to refactor the GameWindow class because I feel like there's much that be moved to separate reusable classes.
* Which part of OpenTK does this affect? Core...?

### Testing status

* Explanation of what’s tested, how tested and existing or new automation tests.
* Can include manual testing by self.
* Specify test plans.
* Rarely acceptable to have no testing.

### Comments

* Any other comments to help understand the change.

### CODE REVIEW

**Comment by DylanFPS** [comment]
Thanks for the contribution, you're awesome! We always welcome contributions from the community. However, there are some points to note about this particular PR:

This will conflict with the work currently going on in #867. Also, to what extent do you want to modularize/subdivide `GameWindow`? We already have `NativeWindow` for all of your non-game needs, and I personally think that it doesn't warrant any more divisions.

Basically, just provide more background and direction in the "Purpose" section of your PR; as you'll need to justify these changes in some cases (such as this one.)

**Comment by daerogami** [comment]
@mtosoftware If you have a proposition for refactoring, present it though the interfaces. If you have an idea on what improvements you would like to see, that would be the best way to propose them. As @DylanFPS mentioned, we are in the process of rewriting the implementation with GLFW using ADL, we are currently re-using the interfaces to maintain as much compatibility as possible.

While I don't want to deter you from making a contribution (we love contributions!), it may be difficult to revise the Game Window pattern while it is currently under re-implementation. If you're still interested, we would love to see what you have in mind!

**Comment by mtosoftware** [comment]
@daerogami One thing I would like to change is right now, we're managing frame updating, frame rendering, also calculating time variables, like deltas and such. This is all happening within the one class.

I propose we create a `GameTime` class, which will be responsible for keeping track of time for the `GameWindow`(and other classes, if needed) **and** it would make the code a little more readable. I found myself jumping back and forth between methods and variables in `GameWindow` just to understand how we keep track of time.

If the implementation is being rewritten, maybe this should be left alone, and I should post an issue?

**Comment by DylanFPS** [comment]
Okay, well make a start and we'll see how it looks! Ideally, we'd not to veer too far away from the current GameWindow implementation, but 4.0 already throws the user into the deep end. So make your changes, and if we think it looks good we'll rebase them onto @daerogami's fork :)

**Comment by DylanFPS** [comment]
Generally, you need to test PRs to make sure everything’s functional but given the circumstances we can probably let this slide (unless we notice something odd in your code)
